### PR TITLE
Create N_bonacci.dart

### PR DESCRIPTION
**Pull request by neha-hasija17**
This Pull Request is for HacktoberFest 2020

# Welcome to Dart community

### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: N_bonacci.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
+void N_bonacci(int n, int m) {
+  var v = new List(m);
```
Comment (Parowicz) [inline N_bonacci.dart:2]: You could return `v` and test it.
```suggestion
import 'package:test/test.dart';

List N_bonacci(int n, int m) {
  List v = new List(m);
```

File: N_bonacci.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  print(v);
+}
+
+void main() {
+  N_bonacci(5, 10);
+  N_bonacci(4, 12);
+  N_bonacci(6, 13);
+  N_bonacci(8, 19);
```
Comment (Parowicz) [inline N_bonacci.dart:20]:
```suggestion
  test("For n=2 N_bonacci is same as fibbonaci", () {
    expect(N_bonacci(2, 6), equals([0, 1, 1, 2, 3, 5]));
  });

  test("For n=3 N_bonacci is same as tribbonaci", () {
    expect(N_bonacci(3, 7), equals([0, 0, 1, 1, 2, 4, 7]));
  });

  test("n=4 N_bonacci", () {
    expect(N_bonacci(4, 10), equals([0, 0, 0, 1, 1, 2, 4, 8, 15, 29]));
  });

  test("n=6 N_bonacci", () {
    expect(N_bonacci(6, 10), equals([0, 0, 0, 0, 0, 1, 1, 2, 4, 8]));
  });

  test("n=8 N_bonacci", () {
    expect(N_bonacci(8, 10), equals([0, 0, 0, 0, 0, 0, 0, 1, 1, 2]));
  });

```

File: N_bonacci.dart [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  for (i = 0; i < m; i++) {
+    v[i] = 0;
+  }
+  v[n - 1] = 1;
+  v[n] = 1;
+  for (i = n + 1; i < m; i++) {
+    v[i] = 2 * v[i - 1] - v[i - 1 - n];
+  }
+
+  print(v);
```
Comment (Parowicz) [inline N_bonacci.dart:13]:
```suggestion
  return v;
```

**Comment by neha-hasija17** [comment]
@Parowicz kindly recheck the code, as the modifications are not accepted by the testers.

**Comment by Parowicz** [comment]
Sry, there was some redundant `\n`-s

**Review by Parowicz — APPROVED** [review]
LGTM!

**Comment by neha-hasija17** [comment]
okay. thank you for merging it

**Pull request merged**